expyfun/stimuli/_tracker.py

```diff
@@ -254,6 +262,31 @@ def respond(self, correct):
             self._x = np.append(self._x, self._x[-1] +
                                 self._current_step_size_up)
 
+        if bound:
+            if self._x_min is not None:
+                if self._x[-1] < self._x_min:
+                    self._limit_count += 1
+                    bad = True
+                    if self._repeat_limit == 'reversals':
+                        reversal = True
+                        self._n_reversals += 1
```

Comment:
you still need to then actually revert the next (currently illegal) step with `self._x[-1] = self._x_min`, no?